sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala

```diff
@@ -261,11 +261,11 @@ class ExplainSuite extends ExplainSuiteHelper with DisableAdaptiveExecutionSuite
             "PartitionFilters: \\[isnotnull\\(k#xL\\), dynamicpruningexpression\\(k#xL " +
               "IN subquery#x\\)\\]"
           val expected_pattern3 =
-            "Location: InMemoryFileIndex \\[.*org.apache.spark.sql.ExplainSuite" +
-              "/df2/.*, ... 99 entries\\]"
+            "Location: InMemoryFileIndex \\[\\S*org.apache.spark.sql.ExplainSuite" +
+              "/df2/\\S*, ... 99 entries\\]"
```

Comment:
I updated this pattern for avoiding a unexpected longest string match in `expected_pattern3`; this PR makes the explain output be placed in a single line because line breaks are removed in `show`. So, the two consecutive `Location:....` strings are regarded as a single match, and then the test in L272 fails.